### PR TITLE
Core done. Working on ext 1

### PR DESCRIPTION
**Pull request by ben-corbin**
I think that the code is slightly broken right now as I'm getting an error in the middle of trying to show movies to only the relevant user.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -26,4 +27,6 @@ model Movie {
   runtimeMins   Int
   createdAt     DateTime    @default(now())
   updatedAt     DateTime    @updatedAt
+  user          User?        @relation(fields: [userId], references: [id])
+  userId        Int?
```
Comment (Willcowen) [inline prisma/schema.prisma:31]: Nice use of an optional one to many, I also did the same to allow for a Movie to not 'need' a user connected to it on creation.

File: src/client/App.jsx [hunk 2]
```diff
@@ -8,22 +8,92 @@ const apiUrl = 'http://localhost:4000';
 function App() {
   const [movies, setMovies] = useState([]);
```
Comment (marianphillips) [inline src/client/App.jsx:10]: The code in this file looks damn nice!

File: src/server/controllers/movie.js [hunk 3]
```diff
@@ -5,22 +5,47 @@ const prisma = new PrismaClient();
 const jwtSecret = 'mysecret';
 
 const getAllMovies = async (req, res) => {
-    const movies = await prisma.movie.findMany();
+    const authorization = req.headers.authorization
+    if (!authorization) {
+        res.status(401)
```
Comment (Willcowen) [inline src/server/controllers/movie.js:10]: Some nice error handling in this function!

File: src/server/controllers/movie.js [hunk 4]
```diff
@@ -5,22 +5,47 @@ const prisma = new PrismaClient();
...
-
+    const authorization = req.headers.authorization
+    if(!authorization) {
+        res.status(401)
+        res.json({ error: "Incorrect login credentials supplied"})
+    }
     try {
-        const token = null;
-        // todo verify the token
+        const token = authorization.split(" ")[1];
```
Comment (Willcowen) [inline src/server/controllers/movie.js:36]: Convert the token string to an Array ['Bearer', 'token'] and take the second item in the array. Awesome!

File: src/server/controllers/movie.js [hunk 5]
```diff
@@ -5,22 +5,47 @@ const prisma = new PrismaClient();
...
+        jwt.verify(token, jwtSecret)
     } catch (e) {
         return res.status(401).json({ error: 'Invalid token provided.' })
     }
 
-    const createdMovie = null;
+    const createdMovie = await prisma.movie.create({
```
Comment (marianphillips) [inline src/server/controllers/movie.js:42]: Could check to see if there are values for all fields. I had some issues (my server crashed) with this.

File: src/client/App.jsx [hunk 6]
```diff
@@ -8,22 +8,92 @@ const apiUrl = 'http://localhost:4000';
...
 
+
   useEffect(() => {
-    fetch(`${apiUrl}/movie`)
+    const options = {
```
Comment (Willcowen) [inline src/client/App.jsx:13]: Interesting use of adding auth to the useEffect, was this to only load the movies for a particular user once logged in?

**Comment by Willcowen** [comment]
Overall, code looks clean and concise, nothing I can fault.

File: src/client/App.jsx [hunk 7]
```diff
@@ -8,22 +8,92 @@ const apiUrl = 'http://localhost:4000';
...
+    }
+    fetch(`${apiUrl}/user/register`, options)
+    .then(res => res.json())
+    .then(json => console.log('Registered user:' + json.data.username))
+    .catch(() => console.log('Failed to register user'))
   };
 
   const handleLogin = async ({ username, password }) => {
     
+    const options = {
```
Comment (marianphillips) [inline src/client/App.jsx:48]: having the options outside the fetch makes it so much easier to read, and something I didnt really consider before. Will do that in future

File: prisma/schema.prisma (hunk 1 above)
Comment (ben-corbin) [reply prisma/schema.prisma:31]: Thanks. I tried it without and was getting an error on migration, this seemed to solve it.

File: src/server/controllers/user.js [hunk 8]
```diff
@@ -2,33 +2,41 @@ const bcrypt = require('bcrypt');
...
 
 const login = async (req, res) => {
     const { username, password } = req.body;
 
-    const foundUser = null;
+    const foundUser = await prisma.user.findFirst({
+        where: { username: username}
+    });
 
     if (!foundUser) {
```
Comment (usamalodhi) [inline src/server/controllers/user.js:29]: good negative checks - where username or password is not found

File: src/client/App.jsx (hunk 6 above)
Comment (ben-corbin) [reply src/client/App.jsx:13]: Yeah that was the idea.

File: prisma/schema.prisma (hunk 1 above)
Comment (Willcowen) [reply prisma/schema.prisma:31]: The same thing happened to me, I wonder if it's needed.

Comment (ben-corbin) [reply prisma/schema.prisma:31]: Some of the other code I've seen this morning doesn't have it so I think not, but it solved my problem and I couldn't see any issue with having it as optional in the schema as checking can be enforced elsewhere if it's mandatory. You should checkout Marian's code for a nice way to make sure the forms are filled properly or get an error.

**Pull request closed without merging**